src/log/file_appender.cpp

```diff
@@ -151,8 +130,9 @@ namespace fc {
    // MS THREAD METHOD  MESSAGE \t\t\t File:Line
    void file_appender::log( const log_message& m )
    {
+      my->rotate_files();
```

Comment:
Perhaps a middle way would be to switch to a new file within `log()` but delete old files in a separate task?

I have tried fiddling around with `yield()` and task priorities but failed to fix it that way. I'm not 100% sure why but I think the problem is that `yield` / `start_next_fiber` will only look at the scheduled tasks queue if no other thread is in the ready list. Modifying that code is something I will try to avoid.

Reply:
> Perhaps a middle way would be to switch to a new file within `log()` but delete old files in a separate task?

Yes, I think this is the best approach at this moment.

Reply:
Done.